sklearnex/ensemble/_forest.py

```diff
@@ -453,16 +453,14 @@ def __init__(
 
         # The estimator is checked against the class attribute for conformance.
         # This should only trigger if the user uses this class directly.
-        if (
-            self.estimator.__class__ == DecisionTreeClassifier
-            and self._onedal_factory != onedal_RandomForestClassifier
+        if self.estimator.__class__ == DecisionTreeClassifier and not isinstance(
+            self._onedal_factory, onedal_RandomForestClassifier
         ):
-            self._onedal_factory = onedal_RandomForestClassifier
-        elif (
-            self.estimator.__class__ == ExtraTreeClassifier
-            and self._onedal_factory != onedal_ExtraTreesClassifier
+            _onedal_factory = onedal_RandomForestClassifier
```

Comment:
Maybe we switch these to a warning? It looks like _onedal_factory would be with these changes only set during the operation of ```__init__```.  Essentially the purpose of these checks is to make sure someone doesn't use a random splitter with random forest, and a decision tree splitter with extra trees if they were to spin their own class based off of ForestRegressor or ForestClassifier